### ISSUE TITLE
Submitting through the Website sets the Time to 12:00

### ISSUE THREAD

**Issue by CryZe** [speedruncom/api#25]
When you submit through the API, the current time is assumed to be the time of the run. This should be customizable in my opinion, but that's not the point of this issue. The issue is that the website has code for showing the times of runs if they didn't happen too long ago, which you can see if you submit through the API. If you submit through the website itself, the run is submitted as 12:00, which makes it so that the following else case never happens:
```php
if ((time() - $sec) > 86400 || date('H:i:s', $sec) == '12:00:00' || !$short) {
    $string = date($dateformat, $localsec);
}
else {
    $string = date(self::$loguser->getTimeFormat(), $localsec);
}
```

**Comment by Lighnat0r** [comment]
Okay so I wrote down some thoughts about this, some of it is a bit of a re-iteration of the issue but just deal with it. Maybe I missed or forgot something in there, let me know, but otherwise this is what it is:

Currently the date is stored as a date-time with the time always set to 12:00 when submitted through the site.
On the other hand, when a run is submitted through the API, the time is set to the current time of day at the moment of submission.

The way I see it, we basically have two options to work away this inconsistency:
1. Get rid of the time altogether and store just the date.
2. Allow the user to set the time when submitting a run through the site (and API).

For submitting through the API, Livesplit already tracks the time of the run, so it would be trivial for them to submit it accurately. 
If/when there are other programs that submit runs using the API, they would be responsible for getting the time somehow. This shouldn't be an issue in most cases, but it might be in some.

When submitting a run through the site, having to enter the time would clutter the submission form, especially considering usually the time will likely not be readily available to the user without more effort to track it.
Having the time optional would save the user effort, but results in the timestamp not being accurate in most cases and thus useless.
If the user submits a video or splits with the run, we could use the time from those (as long as the relevant source has an API that exposes that) but that will not be available in all cases and even if it is, it will not necessarily be accurate (twitch highlights and splits (depending on the source program) might be ok, but youtube videos won't be).

Altogether, I doubt there's a way to get a useable timestamp for runs. Even if there was I think the purpose of storing it would be limited (WR history graph with multiple WR's on one day is the only thing that comes to mind) except for posterity.

As it stands, in my opinion the best option would be to get rid of the time altogether and store just the date.

**Comment by glacials** [comment]
I agree, the value gained from storing time isn't worth the complication. Even if it were, I don't think storing date and time together as a date-time would be the right solution; if it's okay for one of the two to be missing, you're kind of shoved into a corner of representing what should be `null` with something that's not (e.g. 12:00).

**Comment by sgt-kabukiman** [comment]
The API incorrectly used the current time of day when storing the run date. This was a bug either way (my intention was to store 00:00:00), which is now fixed and aligned to the website (i.e. the API also stores 12:00:00).

**Comment by CryZe** [comment]
Ok cool, at least it's consistent now. The problem with not storing the time of day though, is the fact that you can't possibly work with time zones at all. Therefore the run date, based on where you are in the world is always wrong by -1 to +1 days, so it kind of actually is an issue. The thing is that the website most likely shouldn't have a way to put in the time of the run, as that's just too annoying for the users when submitting. It would be cool if the API at least had that, but idk if that's worth the effort.
